### ISSUE TITLE
Failed to Enable Plugin

### ISSUE THREAD

**Issue by andrewchumchal** [sethvargo/vault-secrets-gen#49]
### TL;DR

Failed to enable plugin on version

Version: Vault v1.15.1, built 2023-10-20T19:16:11Z
 Go Version: go1.21.3

### Expected behavior

Enabled Plugin

### Observed behavior

Here's the output

`2023-11-06T15:19:51.279-0400 [WARN]  secrets.secrets-gen.secrets-gen_3b879a3a.secrets-gen: plugin failed to exit gracefully`
`2023-11-06T15:19:51.322-0400 [ERROR] secrets.system.system_9ab6b663: error occurred during enable mount: path=gen/
  error=
  | invalid backend version: 2 errors occurred:
  | \t* fork/exec /etc/vault/plugins/vault-secrets-gen: permission denied
  | \t* fork/exec /etc/vault/plugins/vault-secrets-gen: permission denied
  |`

### Vault CLI (or API) output

```text
$ vault secrets enable \
    -path="gen" \
    -plugin-name="secrets-gen" \
    plugin
```

### Vault server logs

```text
`2023-11-06T15:19:51.279-0400 [WARN]  secrets.secrets-gen.secrets-gen_3b879a3a.secrets-gen: plugin failed to exit gracefully`
`2023-11-06T15:19:51.322-0400 [ERROR] secrets.system.system_9ab6b663: error occurred during enable mount: path=gen/
  error=
  | invalid backend version: 2 errors occurred:
  | \t* fork/exec /etc/vault/plugins/vault-secrets-gen: permission denied
  | \t* fork/exec /etc/vault/plugins/vault-secrets-gen: permission denied
  |`
```
